packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js

```diff
@@ -296,11 +296,11 @@ const defaultTwoColumnModuleLayout = <T: { +[string]: mixed }>({
     if (hasTwoColumnItems) {
       // Currently we only support one two column item at the same time, more items will be supporped soon
       const twoColumnIndex = itemsWithoutPositions.indexOf(twoColumnItems[0]);
+      const isFirstRow = heights.every((height) => height === 0);
 
       // Skip the graph logic if the two column item batch is on the first line
       const skipGraph =
-        heights.every((height) => height === 0) &&
-        itemsWithoutPositions.length <= TWO_COL_ITEMS_MEASURE_BATCH_SIZE;
+        isFirstRow && itemsWithoutPositions.length <= TWO_COL_ITEMS_MEASURE_BATCH_SIZE;
```

Comment:
unrelated to this PR but @manvoid this logic doesn't entirely seem correct. shouldn't this instead be something like
```
const skipGraph = isFirstRow && canFitOnFirstRow?
```

Reply:
Yes, I need to change that, in fact I have another bug if the two column index is on the second batch that means that not all the heights are 0 because the first batch was already positioned, but the two column item is still going to be rendered on the first line. We would need to change the `isFirstRow` to only check if any height is 0 and then check if the tcm can fit there as you mention.

Reply:
i can update this line as part of this PR if you want